### DISCUSSION TITLE
using template with delegate

### DISCUSSION THREAD

**Discussion by teathinker** [skypjack/entt#718] [Q&A] [answered]
Hi, thanks for the excelent framework. I have a question regarding using template with delegate. Following code gives me a compile error like: invalid operands of types ... to binary ‘operator<’. Any idea how to do it? Thanks.

```

tempalte <typename T>
void fun(T& t)
{
     ...
     delegate1.connect<&T::myfun>(t);
}
```

**Comment by skypjack** [comment, marked as answer]
It would help if you could provide a snippet to reproduce the error.
Otherwise, I can only guess and in this case I'd say that the error is due to a missing `template`:
```cpp
delegate1.template connect<&T::myfun>(t);
```

**Reply by teathinker** [reply]
Your guess works. Thanks!
